### ISSUE TITLE
xxxConv object has no attribute '_user_args'

### ISSUE THREAD

**Issue by lina-dev** [pyg-team/pytorch_geometric#7078]
### 🐛 Describe the bug

     script  torch_geometric/nn/conv/message_passing.py
I got the error '`TransformerConv' object has no attribute '_user_args'`

    x = self.conv1(x, edge_index)
  File ".../venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File ".../venv/lib/python3.9/site-packages/torch_geometric/nn/conv/transformer_conv.py", line 177, in forward
    out = self.propagate(edge_index, query=query, key=key, value=value,
  File ".../venv/lib/python3.9/site-packages/torch_geometric/nn/conv/message_passing.py", line 459, in propagate
    coll_dict = self._collect(self._user_args, edge_index, size,
  File ".../venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1269, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'TransformerConv' object has no attribute '_user_args'

      
line 445 (torch-geometric ==**2.3.0**)
```
   if decomposed_layers > 1:
                user_args = **self._user_args**
                decomp_args = {a[:-2] for a in user_args if a[-2:] == '_j'}
                decomp_kwargs = {
                    a: kwargs[a].chunk(decomposed_layers, -1)
                    for a in decomp_args
                }
                decomp_out = []

            for i in range(decomposed_layers):
                if decomposed_layers > 1:
                    for arg in decomp_args:
                        kwargs[arg] = decomp_kwargs[arg][i]

                coll_dict = self._collect(**self._user_args**, edge_index, size,
                                          kwargs)

```

 version **2.2.0** works well, since this version uses correct self.__user_args 
"""line 414
        else:  # Otherwise, run both functions in separation.
            if decomposed_layers > 1:
                user_args = self.__user_args__
                decomp_args = {a[:-2] for a in user_args if a[-2:] == '_j'}
                decomp_kwargs = {
                    a: kwargs[a].chunk(decomposed_layers, -1)
                    for a in decomp_args
                }
                decomp_out = []

            for i in range(decomposed_layers):
                if decomposed_layers > 1:
                    for arg in decomp_args:
                        kwargs[arg] = decomp_kwargs[arg][i]

                coll_dict = self.__collect__(self.__user_args__, edge_index,
                                             size, kwargs)
"""
How to reproduce the error
```
from torch_geometric.nn import GATConv, GCNConv, SAGEConv, TransformerConv
class GraphormerEncoder(torch.nn.Module):
    
    def __init__(self, embedding_size: int = 768, negative_slope=0.2, dropout=0.6, **kwargs):
        
        super().__init__()
        ...
        self.conv1 = TransformerConv(embedding_size, self.hidden_channels, heads, dropout=dropout)
        self.relu = torch.nn.LeakyReLU(negative_slope=negative_slope)
        self.conv2 = TransformerConv(self.hidden_channels * heads, heads, self.final_channels, dropout=dropout)
        
    def forward(self, x, edge_index):
        
        x = self.embedder(x)
        x = self.conv1(x, edge_index)    #<-the error occurs
        x = self.relu(x)
        x = self.conv2(x, edge_index)
        return x



```

### Environment

* PyG version:
* PyTorch version: 
* OS: MAC
* Python version:3.9
* CUDA/cuDNN version:
* How you installed PyTorch and PyG (`conda`, `pip`, source):pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):
torch==1.13.1
more_itertools==9.1.0
datasets==2.10.1
json
pillow==8.4.0
torch-geometric==2.3.0
transformers==4.27.4
layout_lm_inference==0.1.143
botocore==1.29.66
boto3==1.26.66
pytorch-lightning==1.9.1
torch-sparse==0.6.17
torch-scatter==2.1.1

**Comment by rusty1s** [comment]
Mh, I ran your script via
```python
import torch

from torch_geometric.nn import TransformerConv


class GraphormerEncoder(torch.nn.Module):
    def __init__(self, embedding_size: int = 768, negative_slope=0.2,
                 dropout=0.6, heads=1, **kwargs):

        super().__init__()
        self.hidden_channels = 16
        self.final_channels = 16

        self.conv1 = TransformerConv(embedding_size, self.hidden_channels,
                                     heads, dropout=dropout)
        self.relu = torch.nn.LeakyReLU(negative_slope=negative_slope)
        self.conv2 = TransformerConv(self.hidden_channels * heads, heads,
                                     self.final_channels, dropout=dropout)

    def forward(self, x, edge_index):

        x = self.conv1(x, edge_index)
        x = self.relu(x)
        x = self.conv2(x, edge_index)
        return x


model = GraphormerEncoder()

x = torch.randn(10, 768)
edge_index = torch.randint(0, 10, (2, 20))
out = model(x, edge_index)
print(out.size())
```
and it works fine for me. We recently changed some internal names in `MessagePassing`, so this might fail if you are loading an existing model via `torch.load("model_path.pt")`. Is this what you do?

**Comment by lina-dev** [comment]
Hey,
Yes, I load the existing model (torch.load("model_path.pt") ) and then run
an inference code to make predictions.

Best regards, Galina Bakman

чт, 30 мар. 2023 г. в 00:20, Matthias Fey ***@***.***>:

**Comment by rusty1s** [comment]
Yeah, we cannot guarantee that this works across PyG versions (it would mean that we are not allowed to do any modifications to a model). A better approach is to just save the state_dict of a model (see PyTorch docs).
